execution_workflows/IsoSCM/README.md

```diff
@@ -0,0 +1,86 @@
+
+### Parameters
+Parameters used to run the two steps of IsoSCM are specified in conf/modules.config file. 
+Parameters relevant to the workflow itself are:
+- `run_star_genome_generate` - if true, the workflow will run STAR genome generate to obtain genome index. This process takes around 20-25 minutes and only needs to run once per genome. Set to false if the genome index folder already exists.
+- `run_star_alignment` -  if true, the workflow will run STAR alignment to output BAM files with XS strand tags required by IsoSCM
+- `output_dir` - name of the folder that the final output files are going to be in, located under Dapars/results/dapars/
+- `identification_out_suffix` - suffix of the output file for the current run ending with .bed. This will be prefixed with the sample name specified in the sample column of the input sample sheet e.g. samplesheet_example_files.csv
+- `gtf_genome_file` - absolute path to the input GTF annotation file can be obtained by replacing `path_to` with the path to IsoSCM by doing `pwd`, and if using your own genome file, make sure to use the absolute path to your genome file
+- `fasta_genome_file` - absolute path to the input fasta annotation file. If running with APAeval test data, download the fasta genome file by running `wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M18/GRCm38.p6.genome.fa.gz` inside of tests/test_data/ folder.
+- `star_genome_index` - absolute path to the star genome index folder
+
+### Running the identification workflow
+- Change 'identification_out_suffix' parameter in conf/modules.config to the desired file name that ends with '.bed'
+- Change the 'output_dir' to the desired directory name under IsoSCM/results/isoscm the identification challenge output will be in  
+- An example sample sheet is samplesheet_example_files_identification.csv.
+- Run the pilot benchmark nextflow pipeline with nextflow main.nf --input samplesheet_example_files.csv
+
+## Output & post-processing
+By default, each IsoSCM run results in an identification challenge file located under IsoSCM/results/isoscm/challenges_outputs/[sample]_identification_output.bed
+
+## Notes
+- Running STAR genome generate requires 30-40GB of memory
+- Running STAR genome generate takes 20-25 minutes and is only required once per genome
+- IsoSCM doesn't qualify for quantification challenge because IsoSCM assemble step outputs mean read density in the segments upstream and downstream of the polyA site but doesn't output the TPM value for each identified site that is required for quantification challengenge output  
+- IsoSCM doesn't qualify for differential challnege because the compare step provides a confidence score representing the likelihood that a change-point occurs at the given position, compared to a null model that no change-point at this position. However, this doesn't exactly equal to the significance of differential polyadenylation site usage, which is needed for differential challenge output
```

Comment:
I fully agree this means it does not represent significance of differential pA usage. The closest thing to a change in usage metric is the 'differential_usage' column in the output file, but as the README states that is just an effect size difference and no statistical test is performed on the significance of that change in usage (which is what we want to assess with the differential usage challenge). 

Also seems like `compare` can only handle comparisons between 2 samples (i.e. it cannot handle replicates), which is another factor that makes it difficult to adapt to our datasets (author suggests merging all replicates into a single BAM file https://github.com/shenkers/isoscm/issues/8). I believe lack of ability to handle replicates for differential usage may have been one of the reasons we haven't pressed ahead with MISO).

Apologies for a bit of a ramble, feel like I'm not really suggesting anything. Maybe just to clarify that isoSCM doesn't perform any statistical tests on the significance of the difference in polyA site usage. Feel free to ignore this haha

Reply:
Thank you for spilling your thoughts on this! I took some time thinking about this trying to see if we can include isoscm in differential challenge by using any of the outputs they provide so it's nice to have an extra confirmation from you! And yes you're right compare can only handle comparisons between 2 samples and no replicates so it would be difficult to use with our datasets. It might be doable by adding an extra process in the workflow but since it doesn't output significance that we need anyway so phewh not a worry for now :p 

But yes in summary what I've found is isoscm doesn't perform any statistical test on the significance of the difference in polyA site usage and I confirmed this with the author too :)